src/drivers/imx/dummy-dma.c

```diff
@@ -3,28 +3,116 @@
 // Copyright 2019 NXP
 //
 // Author: Daniel Baluta <daniel.baluta@nxp.com>
+// Author: Paul Olaru <paul.olaru@nxp.com>
 
+#include <sof/atomic.h>
 #include <sof/lib/dma.h>
+#include <errno.h>
 #include <stdint.h>
+#include <sof/string.h>
+#include <config.h>
+#include <sof/audio/component.h>
 
-/* allocate next free DMA channel */
+#define trace_dummydma(__e, ...) \
+	trace_event(TRACE_CLASS_DMA, __e, ##__VA_ARGS__)
+#define tracev_dummydma(__e, ...) \
+	tracev_event(TRACE_CLASS_DMA, __e, ##__VA_ARGS__)
+#define trace_dummydma_error(__e, ...) \
+	trace_error(TRACE_CLASS_DMA, __e, ##__VA_ARGS__)
+
+/* Dummy DMA driver (software-based DMA controller)
+ *
+ * This driver is usable on all platforms where the DSP can directly access
+ * all of the host physical memory (or at least the host buffers).
+ *
+ * The way this driver works is that it simply performs the copies
+ * synchronously within the dma_start() and dma_copy() calls.
+ *
+ * One of the drawbacks of this driver is that it doesn't actually have a true
+ * IRQ context, as the copy is done synchronously and the IRQ callbacks are
+ * called in process context.
+ *
+ * An actual hardware DMA driver may be preferable because of the above
+ * drawback which comes from a software implementation. But if there isn't any
+ * hardware DMA controller dedicated for the host this driver can be used.
+ *
+ * This driver requires physical addresses in the elems. This assumption only
+ * holds if you have CONFIG_HOST_PTABLE enabled, at least currently.
+ */
+
+struct dma_chan_pdata {
+	struct dma_sg_elem_array *elems;
+	int sg_elem_curr_idx;
+	uintptr_t r_pos, w_pos;
+};
+
+/* allocate next free DMA channel; we will ignore req_chan because
+ * channel numbers are not meaningful
+ */
 static struct dma_chan_data *dummy_dma_channel_get(struct dma *dma,
```

Comment:
Let's use kernel-doc for commenting functions. https://www.kernel.org/doc/html/v4.10/process/coding-style.html#commenting